testy/tagged.go

```diff
@@ -0,0 +1,96 @@
+package testy
+
+import (
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	TagUnit        = "unit"
+	TagIntegration = "integration"
+	TagPostgres    = "postgres"
+	TagRabbit      = "rabbit"
+)
+
+// TaggedTestsEnvVar defines the name of the environment variable for the tagged tests.
+// Example:
+// 		TaggedTestsEnvVar="TEST_TAGS"
+//		env TEST_TAGS="unit" go test ./...
+var TaggedTestsEnvVar = "TEST_TAGS"
+
+// TaggedTest runs the test if the provided tag matches at least one runtime tag.
+// Example:
+// 		func TestSomething(t *testing.T) {
+//    		TaggedTest(t, "unit")
+//    		...
+// 		}
+// Run with:
+// 		env TEST_TAGS="unit,integration" go test ./...
+func TaggedTest(t *testing.T, testTag string) {
+	tags := getRuntimeTags()
+	if shouldRun := tags.empty() || tags.contains(testTag); !shouldRun {
```

Comment:
open question: do you think we should run the test when `tags.empty()`? I actually prefer not running the test on `tags.empty()` because I think the opt-in flow is better especially for a new engineer:

Opt-out flow (execute test on `tags.empty()`):
- New engineer run `go test ./...`, all tests are run
- He start seeing errors everywhere because database, queues, etc are not set up properly
- Either read the test code/ask a teammate/check documentation, to discovers the TEST_TAGS env variable
- he tries once again with the correct command

Opt-in flow (do NOT execute test on `tags.empty()`):
- New engineer run `go test ./...`, all tests are skipped
- He sees the skip message "run with unit TEST_TAGS to run this test" (that's why IMO it is important to give the skip reason by using `t.Skipf()`
- He gave another try `TEST_TAGS=unit go test ./...`, succeeded
- New engineer also understands naturally for `postgres` test, the DB needs to be set up first